spec/controllers/api/v1/decks_controller_spec.rb

```diff
@@ -0,0 +1,427 @@
+        expect(json_response.key?('updated_at')).to eq(false)
+      end
+    end
+  end
+  
+  describe 'PATCH #update' do
+    let!(:deck) { FactoryBot.create(:deck) }
+    
+    context 'with valid params' do
+      before :each do
+        @decks_before_request = Deck.count
+        @old_title = deck.reload.title
+        @deck_params = FactoryBot.attributes_for(:deck)
+        patch :update, params: { id: deck.id, deck: @deck_params }
+      end
+      
+      it 'is expected to have :ok (200) HTTP response code' do
+        expect(response.status).to eq(200)
+      end
+      
+      it 'is expected to return application/json content_type' do
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+      end
+      
+      it 'is expected to update fields for Deck' do
+        deck.reload
+        @deck_params.each do |key, value|
+          expect(deck[key]).not_to eq(@old_title)
+          expect(deck[key]).to eq(value)
+        end
+      end
+      
+      it 'is expected to NOT create a new Deck' do
+        expect(Deck.count).to eq(@decks_before_request)
+      end
+      
+      it 'is expected to include fields' do
+        deck = JSON.parse(response.body)['deck']
+        expect(deck.key?('id')).to eq(true)
+        expect(deck.key?('title')).to eq(true)
+      end
+      
+      it 'is expected to NOT include fields' do
+        deck = JSON.parse(response.body)['deck']
+        expect(deck.key?('created_at')).to eq(false)
+        expect(deck.key?('updated_at')).to eq(false)
+      end
+    end
+    
+    context 'Deck NOT found' do
+      before :each do
+        patch :update, params: { id: -1 }
+      end
+      
+      it 'is expected to have :not_found (404) HTTP response code' do
+        expect(response.status).to eq(404)
+      end
+      
+      it 'is expected to return application/json content type' do
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+      end
+      
+      it 'is expected to return errors' do
+        json_response = JSON.parse(response.body)
+        expect(json_response.key?('errors')).to eq(true)
+      end
+    end
+  end
+  
+  describe 'PUT #update' do
+    let!(:deck) { FactoryBot.create(:deck) }
+    
+    context 'with valid params' do
+      before :each do
+        @decks_before_request = Deck.count
+        @old_title = deck.reload.title
+        @deck_params = FactoryBot.attributes_for(:deck)
+        patch :update, params: { id: deck.id, deck: @deck_params }
+      end
+      
+      it 'is expected to have :ok (200) HTTP response code' do
+        expect(response.status).to eq(200)
+      end
+      
+      it 'is expected to return application/json content_type' do
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+      end
+      
+      it 'is expected to update fields for Deck' do
+        deck.reload
+        @deck_params.each do |key, value|
+          expect(deck[key]).not_to eq(@old_title)
```

Comment:
This is garbage. Why are you checking all values against value for just `title`?